src/excalidraw-backend/server.ts

```diff
@@ -311,18 +316,25 @@ export class Server {
       for await (const _ of setInterval(this.saveIntervalMs, null, {
         signal: ac.signal,
       })) {
-        const saved = await this.sendSaveMessage(roomId, {
+        const data = await this.sendSaveMessage(roomId, {
           timeout: this.saveTimeoutMs,
         });
 
-        if (saved === undefined) {
+        if (data === undefined) {
           this.logger.verbose?.(
             `No eligible sockets found to save '${roomId}'`,
           );
-        } else if (saved) {
-          this.logger.verbose?.(`Saving '${roomId}' successful`);
+        } else if (data.saved) {
+          this.logger.verbose?.(
```

Comment:
you are not resetting `failedSaves` to 0 here, which is fine, but then it's not `consecutiveFailedAttempts` should be just `failedAttempts`... Maybe I'm wrong and it's resetted somewhere else, but I can't see where

Reply:
After the limit is reached the flag is `false` and the socket is filtered out. For the lifetime of it's session it can no longer be picked.
This is reset when a new session (connection) is established. In theory it's never reset, because a brand new socket is initialized, but in practise this behaviour serves that purpose.